src/pages/api/repositories/baseRepository.ts

```diff
@@ -0,0 +1,28 @@
+async function handleAsync<T>(operation: () => Promise<T>): Promise<T> {
+  try {
+      return await operation();
+  } catch (error) {
```

Comment:
Unnecessary Try-catch, if you are just throwing everything again. Just let the calling function catch the error.